kong/db/migrations/core/016_280_to_300.lua

```diff
@@ -165,6 +184,8 @@ local function c_migrate_regex_path(coordinator)
         end
       end
 
+      validate_atc_expression(route)
```

Comment:
This might not be good enough, as the bad expression does not prevent the migration from continuing, and is already too late for user to fix it.

Reply:
Do we have `dry_run` or `preflight` mode?

Reply:
Should we stop the user's migration progress? It may be too offensive.

Reply:
The process needs to be stopped until the user fixes the issue. Otherwise user won't be able to start Kong again with the bad expression.

We should do the check and stop the user before altering any table.

Reply:
I changed the validation logic, now we check all routes before the actual db action.